### PR TITLE
Use dask to parallelize chunk computation as much as possible.

### PR DESCRIPTION
**Pull request by stefanseefeld**
This is work in progress...

### CODE REVIEW

**Comment by kwmsmith** [comment]
@stefanseefeld is this still work-in-progress?

**Comment by stefanseefeld** [comment]
Please consider this PR complete.

File: odo/chunks.py [hunk 1]
```diff
@@ -34,8 +35,16 @@ def __init__(self, data):
     def __iter__(self):
         if callable(self.data):
             return self.data()
-        else:
-            return iter(self.data)
+        elif isinstance(self.data, list) and len(self.data) and callable(self.data[0]):
```
Comment (kwmsmith) [inline odo/chunks.py]: nit: could you break up this test over several lines for readability?

File: odo/convert.py [hunk 2]
```diff
@@ -207,34 +208,43 @@ def _():
...
+        if not data:
+            data = [convert(pd.DataFrame, [], **kwargs)]
+        return chunks(pd.DataFrame)(data)
+
+    # TODO: Decide whether we should support the `add_index` flag at all.
+    #       If not, the following code can be removed (and an exception be
+    #       raised.
+    #
+    #else:
+    #    raise ValueError("The 'add_index' flag is not supported by this backend.")
```
Comment (kwmsmith) [inline odo/convert.py]: Can we keep the TODO comment but remove this commented out `else` clause?

**Comment by kwmsmith** [comment]
Some nits, but otherwise LGTM.

**Comment by kwmsmith** [comment]
Please add an entry to the latest `whatsnew` file for this improvement before merging.

**Comment by stefanseefeld** [comment]
While adding a release note (using a new `whatsnew/0.6.0.txt` file) I noticed that the https://github.com/blaze/odo/blob/master/docs/source/releases.rst file is very much out of date, as the last entry is from the 0.3.4 release. Addressing this however seems out of scope for this PR.

**Comment by llllllllll** [comment]
Could you open an issue about this. These pull requests were merged a long time ago.

**Comment by Matttman** [comment]
There is an issue open for it from a few months ago. #585

**Pull request merged**